packages/node/src/integrations/tracing/vercelai/index.ts

```diff
@@ -0,0 +1,195 @@
+/* eslint-disable complexity */
+import { SEMANTIC_ATTRIBUTE_SENTRY_OP, defineIntegration, spanToJSON } from '@sentry/core';
+import type { IntegrationFn } from '@sentry/types';
+import { generateInstrumentOnce } from '../../../otel/instrument';
+import { addOriginToSpan } from '../../../utils/addOriginToSpan';
+import { SentryVercelAiInstrumentation } from './instrumentation';
+
+const sentryVercelAiInstance = new SentryVercelAiInstrumentation({});
+
+export const instrumentVercelAi = generateInstrumentOnce('vercelAI', () => sentryVercelAiInstance);
+
+const _vercelAIIntegration = (() => {
+  return {
+    name: 'vercelAI',
+    setupOnce() {
+      instrumentVercelAi();
+    },
+    preprocessEvent(event) {
+      if (!sentryVercelAiInstance.patchIsActive) {
+        return;
+      }
+
+      if (event.type === 'transaction' && event.spans?.length) {
+        for (const span of event.spans) {
+          const { data: attributes, description: name } = span;
+
+          if (!attributes || !name || span.origin !== 'auto.vercelai.otel') {
+            continue;
+          }
+
+          // attributes around token usage can only be set on span finish
+          span.data = span.data || {};
+
+          if (attributes['ai.usage.completionTokens'] != undefined) {
+            span.data['ai.completion_tokens.used'] = attributes['ai.usage.completionTokens'];
+          }
+          if (attributes['ai.usage.promptTokens'] != undefined) {
+            span.data['ai.prompt_tokens.used'] = attributes['ai.usage.promptTokens'];
+          }
+          if (
+            attributes['ai.usage.completionTokens'] != undefined &&
+            attributes['ai.usage.promptTokens'] != undefined
+          ) {
+            span.data['ai.tokens.used'] = attributes['ai.usage.completionTokens'] + attributes['ai.usage.promptTokens'];
```

Comment:
looking at our [LLM monitoring attribute spec](https://develop.sentry.dev/sdk/telemetry/traces/modules/llm-monitoring/#span-data), should this rather be:

```suggestion
            span.data['ai.total_tokens.used'] = attributes['ai.usage.completionTokens'] + attributes['ai.usage.promptTokens'];
```

? Not sure though, this is the first time I'm looking at LLM monitoring